crates/core/src/inline_snippets.rs

```diff
@@ -300,6 +300,27 @@ fn delete_hanging_comma(
     replacements: &mut [(EffectRange, String)],
     offset: usize,
 ) -> Result<(String, Vec<ReplacementInfo>)> {
+    // Handle the case when after applying replacements, a single comma
```

Comment:
@morgante hey, i tried with your given pattern on [https://app.grit.io/studio?key=-Pm14wW6mvH0S41J_hFh8](https://app.grit.io/studio?key=-Pm14wW6mvH0S41J_hFh8), it worked. 

Since I am still new to gritql and its rules, can you give a example where it fails. Do you mean presence of matching keywords on multiple lines ? or presence of keyword on the first line (that works) ..

Reply:
@morgante 
you mean when it is like this:
```
bibtex_citation="",
n_samples={_EVAL_SPLIT: 1820}, avg_character_length={_EVAL_SPLIT: 78},
```

then my current fix prints with a extra space:
```
bibtex_citation="",
 stats=GeneralDescriptiveStats(n_samples={_EVAL_SPLIT: 1820}, avg_character_length={_EVAL_SPLIT: 78}),
```

Reply:
I don't find any other edge cases,
Since I am still new to gritql, can you provide a case where it fails.

Reply:
Your fix completely fails in cases where the arguments are on 1 line. *The original bug isn't related to line handling at all and introducing that is a fundamental mistake in your fix.*

```

/// Same as above test, but ensures the behavior doesn't depend on line breaks
#[test]
fn trailing_comma_after_argument_removal_one_line() {
    run_test_expected({
        TestArgExpected {
            pattern: r#"
                language python
                `TaskMetadata($args)` where {
                        $args <: any {
                            contains `n_samples=$_` as $ns_kwarg where {
                                $ns_kwarg <: `n_samples = $ns_val` => .
                            },
                            contains `avg_character_length=$_` as $avg_kwarg where {
                                $avg_kwarg <: `avg_character_length = $avg_val` => `stats=GeneralDescriptiveStats(n_samples=$ns_val, avg_character_length=$avg_val)`
                            },
                        },
                    }
                "#.to_owned(),
            source: r#"
                TaskMetadata(description="Parallel news titles from the Tbilisi City Hall website (https://tbilisi.gov.ge/).", main_score="f1", domains=["News"], text_creation="created", n_samples={_EVAL_SPLIT: 1820}, reference="https://huggingface.co/datasets/jupyterjazz/tbilisi-city-hall-titles")
            "#
                .to_owned(),
            expected: r#"
                TaskMetadata(description="Parallel news titles from the Tbilisi City Hall website (https://tbilisi.gov.ge/).", main_score="f1", domains=["News"], text_creation="created",   reference="https://huggingface.co/datasets/jupyterjazz/tbilisi-city-hall-titles")
            "#
                .to_owned(),
        }
    })
    .unwrap();
}
```

Also has nothing to do with GritQL, you don't need to change the pattern to see how this is broken.